test/projection.jl

```diff
@@ -170,10 +170,36 @@ end
         @test x == ProjectTo(x)(@thunk(ZeroTangent()))
     end
 
-    @testset "to Transpose" begin # TODO: this one, plus Adjoint, and SubArray
+    @testset "to Transpose" begin
         x = rand(3, 4)
         t = transpose(x)
-        
-        #@test x == ProjectTo(t)(rand(4, 3))
+        mt = collect(t)
+        a = adjoint(x)
+        ma = collect(a)
+
+        @test t == ProjectTo(t)(mt)
+        @test t == ProjectTo(t)(ma)
+        @test zeros(4, 3) == ProjectTo(t)(ZeroTangent())
+        @test zeros(4, 3) == ProjectTo(t)(Tangent{Transpose}(; parent=ZeroTangent()))
+    end
+
+    @testset "to Adjoint" begin
```

Comment:
It's probably important to test this on arrays with complex numbers.
Since that is when it differs from transpose